File: 1013-FibonacciNumber/1013-FibonacciNumber.java
// Last updated: 1/3/2026, 6:25:50 PM
class Solution {
    public int fib(int n) {
     if(n <= 1){
        return n;
     }
     int last = fib(n-1);
     int secLast = fib(n-2);
     return last + secLast;
    }
}

File: 1013-FibonacciNumber/1013-FibonacciNumber.cpp
// Last updated: 1/3/2026, 6:25:46 PM
class Solution {
public:
    int fib(int n) {
        if(n == 0) return 0;
        if(n == 1) return 1;
        int ans = 0;
        int first = 0;
        int second = 1;
        for (int i = 2; i <= n; i++) {
            ans = first + second;
            first = second;
            second = ans;
        }
        return second;
    }
};